diff --git a/neuston_data.py b/neuston_data.py
index 6ab77eb..56d41fb 100644
--- a/neuston_data.py
+++ b/neuston_data.py
@@ -448,10 +448,19 @@ class IfcbBinDataset(Dataset):
         else:
             bin_images = bin.images
 
-        for target_number, img in bin_images.items():
-            target_pid = bin.pid.with_target(target_number)
-            self.images.append(img)
-            self.pids.append(target_pid)
+        try:
+            for target_number, img in bin_images.items():
+                try:
+                    target_pid = bin.pid.with_target(target_number)
+                except ValueError as e:
+                    print(f'Error processing {bin.pid} target {target_number}, skipping: {e}')
+                    continue
+
+                self.images.append(img)
+                self.pids.append(target_pid)
+        except Exception as e:
+            print(f'Error processing {bin.pid}: {e}')
+
 
     def __getitem__(self, item):
         img = self.images[item]
